packages/camera/camera_android_camerax/example/integration_test/integration_test.dart

```diff
@@ -178,4 +179,86 @@ void main() {
       }
     }
   });
+
+  testWidgets('Video capture records valid video', (WidgetTester tester) async {
+    final List<CameraDescription> cameras = await availableCameras();
+    if (cameras.isEmpty) {
+      return;
+    }
+
+    final CameraController controller = CameraController(cameras[0],
+        mediaSettings:
+            const MediaSettings(resolutionPreset: ResolutionPreset.low));
+    await controller.initialize();
+    await controller.prepareForVideoRecording();
+
+    await controller.startVideoRecording();
+    final int recordingStart = DateTime.now().millisecondsSinceEpoch;
+
+    sleep(const Duration(seconds: 2));
+
+    final XFile file = await controller.stopVideoRecording();
+    final int recordingTime =
+        DateTime.now().millisecondsSinceEpoch - recordingStart;
```

Comment:
I think we can make this test a bit better by comparing on both sides of the stopRecording call. 


```suggestion
    // Test should pass with this at any non zero value. 
    sleep(const Duration(seconds: 2));
    
     final int preStopTime =
        DateTime.now().millisecondsSinceEpoch - recordingStart;
    final XFile file = await controller.stopVideoRecording();
    final int postStopTime =
        DateTime.now().millisecondsSinceEpoch - recordingStart;
        ....
        expect(duration, greaterThan(preStopTime));
        expect(duration, lessThan(postStopTime));
```

Note: 
We should be using a package like https://pub.dev/packages/system_clock because DateTime.now is vulnerable to the clock changing while the test is running but that feels out of scope.

Reply:
The preStopTime check didn't quite work I assume because stopping video recording takes so long but I kept the naming of postStopTime for clarity

Reply:
I suppose we could also fix this by listening for the [`VideoRecordEvent.Start`](https://developer.android.com/reference/androidx/camera/video/VideoRecordEvent.Start) event to return. Let me give it a go!

Reply:
@reidbaker Ok so I added the fix but it didn't quite work still :/ the duration increased but guessing there still some lag time in starting the video. Still left it in though as it is a better approximation.

Reply:
Going to land this as-is since I've done all I can to remedy this issue. If we feel like we need this fixed, I can file a bug and escalate to CameraX.

Reply:
Filed https://github.com/flutter/flutter/issues/148138 for this.